[게시글 생성/조회/수정/삭제] 코맥 미션 제출합니다.

parker1609 opened the pull request:
스프링부트를 처음하여 많이 부족합니다. 
미션을 수행하면서 이렇게 구현하는게 맞는지 몰랐던 부분이 많았습니다.
많은 지적 부탁드립니다!

school0bhy committed:
docs : README.md 작성

school0bhy committed:
feat: 게시글 작성 페이지 이동 구현

* index.html, article-edit.html 파일을 templates 폴더로 이동함

* AriticleController.java: /writing GET요청 처리

Squashed commit of the following:

commit 7fa33210c68665351585da9f8fbc4e2117248a2e
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Tue Jul 9 17:26:55 2019 +0900

    docs: TODO 추가

commit d8e14dbdd71112723278e2e312ec46582a131d00
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Tue Jul 9 17:26:17 2019 +0900

    feat: 게시글 작성 페이지 이동

school0bhy committed:
feat: 게시글 작성 및 추가 기능 구현

school0bhy committed:
feat: 게시글 목록 조회

commit 9361045bbb8cfc607918c5ed4bda4ceee9aece6f
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Wed Jul 10 18:01:03 2019 +0900

    feat: 게시글 목록 조회

school0bhy committed:
feat: 게시글 조회 기능

Squashed commit of the following:

commit 513b0f9f8dcf8cd2a2810db7d8e82c3efd13b317
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 08:42:08 2019 +0900

    feat: 게시글 조회 기능 구현

commit 55b1b0ea2760d83c12b1cc060e2fa0d03e9d8297
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 08:29:05 2019 +0900

    chore: 절대경로로 변경

    * 외부 리소스 접근 경로 상대경로에서 절대경로로 변경

school0bhy committed:
feat: 게시글 수정 기능 구현

school0bhy committed:
feat: 게시글 삭제 기능 구현

Squashed commit of the following:

commit 44fa3af15a2b1b6da8b8229f06a26c315cfa493c
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 14:11:07 2019 +0900

    docs: TODO.md 추가

commit b01f06e590c2b8b42a40c8da36adb01705bfbf7f
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 14:01:39 2019 +0900

    refactor: html js부분 중복 제거

commit 51101469f093fd305653c6ed63d5be8aaa69ad24
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:52:07 2019 +0900

    fix: article-edit.html 나가기 버튼 오류 수정

commit 23ba3a40a3c427248c4ee8f1bdbf16a1119d1798
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:49:11 2019 +0900

    feat: 게시글 삭제 시 redirect

commit 70d5a2e9232f2baea54bbe3ac554696b89ad9149
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:29:53 2019 +0900

    feat: 게시글 삭제 후 남은 게시글 보여주기

commit 184ddc4acf42506d2f46d48eb4e79b4dae6173ce
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:21:21 2019 +0900

    refactor: 들여쓰기

commit 4e8b30e3dfa72ed4babfba2f4e3ffd9b571f8787
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 09:16:19 2019 +0900

    feat: 게시글 삭제 기능 구현

commit 5111c6e2d26adec5799a4546a3100a0e3f874835
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 01:43:53 2019 +0900

    refactor: Article 클래스 id 필드 추가

commit 2e71b5f6a5010bbe5cecca9a4f26616f53788450
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Fri Jul 12 01:42:44 2019 +0900

    chore: html 중복 제거

commit b88d426c35f40c12d1e069b9480789d0c7937563
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 16:54:29 2019 +0900

    refactor: UI 버그 수정

    * app.css, index.html 수정

commit 7c80edaf0f4452a5f610487086c466ce0da8292a
Author: school0bhy <house1217@ajou.ac.kr>
Date:   Thu Jul 11 16:49:31 2019 +0900

    docs: 게시글 삭제 기능 TODO 작성

school0bhy committed:
refactor: InvalidArticleException 로 Article 예외 처리

kang-hyungu reviewed (changes requested):
안녕하세요 코맥님 리뷰어 강현구입니다.
스프링 부트가 처음이라고 하셨는데 잘 구현하셨어요 :+1: 
지난 과정에서 학습한대로 도메인 클래스에 테스트 작성한 것도 아주 좋습니다 :100: 
몇 가지 피드백 남겼으니 확인해보아요

parker1609 committed:
refactor: implementation 키워드 사용

parker1609 committed:
refactor: ArticleNotFoundException 추가

parker1609 committed:
refactor: 게시글 수정 시 원본 게시글 내용이 변경되도록 변경

parker1609 committed:
refactor: constructor injection으로 변경

parker1609 committed:
refactor: toArticle() 메서드 추가

parker1609 committed:
refactor: find()에서 findById()로 이름 변경

parker1609 committed:
refactor: 게시글 생성 중복 제거

parker1609 committed:
refactor: 게시글 생성 및 삭제 중복 제거

parker1609 committed:
refactor: 게시글 수정이 정상적으로 수행되었는지 확인

parker1609 committed:
refactor: html 중복 제거

parker1609 committed:
refactor: 그 외 피드백 적용

Squashed commit of the following:

commit 23351a2f8aa281f72d70d94ad5ed89d9c996316d
Author: CODEMCD <psbum1609@gmail.com>
Date:   Sun Jul 14 13:50:22 2019 +0900

    refactor: id 동기화

commit 65fa1ca46087f5b47b85742cd13c871b5f8e197a
Author: CODEMCD <psbum1609@gmail.com>
Date:   Sun Jul 14 13:02:19 2019 +0900

    refactor: RestController 사용

commit 6ae60f47ecc6ed287621da5499d0f9f8aa43bb2b
Author: CODEMCD <psbum1609@gmail.com>
Date:   Sun Jul 14 12:49:41 2019 +0900

    refactor: Collections을 반환할 때 변경 불가하게 수정"

kang-hyungu reviewed (approved):
반영하신 내용 확인했습니다!
피드백 깔끔하게 잘 적용하셨네요 :+1: 
코드 구현하느라 수고하셨습니다
